onnxruntime/core/providers/cpu/reduction/reduction_kernel_base.h

```diff
@@ -0,0 +1,40 @@
+#ifndef CORE_PROVIDERS_CPU_REDUCTION_KERNEL_BASE_H
```

Comment:
Was it necessary to create a separate file for this one?

Reply:
The original header contains much implementation. It is include from cuda kernel implementation. Compilation of the cuda implementation breaks with new changes to the header. The cuda implementation only needs the bast class, not the majority of the rest. It is better to have a base header shared by both cpu and cuda.